src/raft_log.rs

```diff
@@ -310,7 +316,11 @@ impl<T: Storage> RaftLog<T> {
         if idx == 0 {
             return;
         }
-        if idx > cmp::min(self.committed, self.persisted) || idx < self.applied {
+        // Do not check idx with committed or persisted index here becase when `apply_unpersisted_log_limit` > 0:
+        // 1. then it is possible idx > persisted.
+        // 2. when the application restart after applied but before committed entried(and committed index) is persisted
+        //    then it is also possible idx > committed.
+        if idx < self.applied {
```

Comment:
At least `idx` should be greater than or equal to `self.committed`.

Reply:
This is for the initialization phase, currently it call applied_to to set the applied_index. I will change it to keep this check for normal path.